packages/flutter/lib/src/painting/path_border.dart

```diff
@@ -0,0 +1,191 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+
+import 'border_radius.dart';
+import 'borders.dart';
+import 'box_border.dart';
+import 'edge_insets.dart';
+
+/// The signature of a method that creates a [Path] for a [PathBorder].
+///
+/// The `bounds` parameter will not be null.
+typedef PathBorderBuilder = Path Function(Rect bounds, TextDirection textDirection);
+
+/// A [BoxBorder] that delegates painting its border to a [PathBorderBuilder].
+///
+/// This class can be used draw a customized border as a [BoxDecoration], for
+/// example a path that has been dashed or trimmed.
+///
+/// {@tool sample}
+/// This will draw a container with a red elliptical border:
+///
+/// ```dart
+/// Container(
+///   decoration: BoxDecoration(
+///     border: PathBorder(
+///       pathBuilder: (Rect bounds, TextDirection textDirection) =>
+///           Path()..addOval(bounds),
+///       border: const BorderSide(color: Colors.red),
+///     ),
+///   ),
+///   padding: const EdgeInsets.all(20.0),
+///   child: const Text('You have pushed the button this many times:'),
+/// );
+/// ```
+/// {@end-tool}
+///
+/// [PathBorders] are defined as being uniform regardless of inputs, although
+/// there is no constraint in this class forcing a uniform path to be drawn by
+/// the [PathBorderBuilder].
+class PathBorder extends BoxBorder {
```

Comment:
i hadn't realised this extends BoxBorder. That makes even less sense to me. this implements even less of the BoxBorder API than it does the Shape API.

Reply:
Ahhhh ok.  I think I can achieve what I want with a different class.  I'm going to close this and see if I really need it or if it can just be reworked.